### PR TITLE
Provide capability to use an existing Kubernetes secret for the gatew…

### LINKED ISSUE #84: Provide capability to use an existing Kubernetes secret for the gateway.auth.key
### Proposal

Right now we need to input the gateway.auth.key as a value in the helm chart. And the secret is created by the helm chart
However it would be nice to also be able to reference an existing secret. 

### Use-Case

This way the secret value can be managed separately from the helm chart.
Also the key is not visible, only the secret name.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by LDL-GH**
…ay.auth.key
closes  #84
Signed-off-by: Laurens Peeters <laurens.peeters@delijn.be>

### CODE REVIEW

File: helm-charts/azure-api-management-gateway/values.yaml [hunk 1]
```diff
@@ -17,8 +17,15 @@ gateway:
...
     uri: 
   auth:
+    # When an existing secret is configured, the value specified here will not be used
     key: 
 
+secret:
+  useExisting:
+    enabled: false
+    name: 
+
```
Comment (tomkerkhove) [inline helm-charts/azure-api-management-gateway/values.yaml:27]: Let's keep it flat and use `secret.name` when `secret.create: true`
```suggestion
secret:
  create: true
  name: 

```

**Review by tomkerkhove — APPROVED** [review]

File: helm-charts/azure-api-management-gateway/values.yaml (hunk 1 above)
Comment (LDL-GH) [reply helm-charts/azure-api-management-gateway/values.yaml:27]: Using this syntax, wouldn't it be confusing to only use the name when using an existing secret.
Do I use secret.name value in the secret created by the helm chart as well?
With secret.name defaulting to "azure-api-management-gateway" in values.yaml?

Comment (tomkerkhove) [reply helm-charts/azure-api-management-gateway/values.yaml:27]: Fair point, what about this:
```suggestion
secret:
  createSecret: true
  secretName: ""

```

That way it's more clear and flat, thoughts?

Comment (tomkerkhove) [reply helm-charts/azure-api-management-gateway/values.yaml:27]: I'm happy to jump in and help on this reshuffling though, if you want.

Comment (LDL-GH) [reply helm-charts/azure-api-management-gateway/values.yaml]: The suggested change should be implemented

**Review by tomkerkhove — APPROVED** [review]
LGTM, thank you!

**Pull request merged**